### ISSUE TITLE
Linter: build a way to define the path to test 

### ISSUE THREAD

**Issue by camc314** [oxc-project/oxc#1930]
Several of the eslint plugin next (#1929) require us to know the path (are we in app dir/pages dir/other ect)

e.g. https://github.com/vercel/next.js/blob/canary/packages/eslint-plugin-next/src/rules/no-head-element.ts#L20-L24

We should build a way to test these properly.

probably something like
```rs
let pass = vec![
    ("source", Option<config>, Option<meta>)
]
```

Where `meta` is a struct that can contain the file path - it could be expanded for other data in the future

**Comment by Boshen** [comment]
What about adding a new API? My decision with tuples was stupid.

```
struct TestCase {
  source: String,
  config: Option<Config>,
  path: PathBuf
}
```

We'll then add a bunch of different constructors for it.

For supporting existing cases, a `From` implementation on `Tester::new` should make it work.

**Comment by camc314** [comment]
yeah that seems reasonable

**Comment by camc314** [comment]
Closed by #1955
